### DISCUSSION TITLE
Lock row / Disable delete for specific row

### DISCUSSION THREAD

**Discussion by Kenny407** [nick-keller/react-datasheet-grid#125] [Q&A] [answered]
Hello, 

First of all thanks for this awesome package, we use it in our company and we are very happy with how fast it is and the community around it.

I'm currently trying to lock the right click actions on a specific row / avoid deleting the row but still be able to add more rows.

My use case is that I'm displaying a grid with some rows that contain user information, but depending on certain criteria one user should not disappear from the grid. Ideally it would be greyed out (similar to `disabled` prop for columns) just that for rows.

Do you think it's feasible or can it be accomplished via another way?

Thanks in advance

**Comment by nick-keller** [comment, marked as answer]
Hello @Kenny407 
- You can disable a cell based on a condition (just pass a function instead of a boolean)
- You can do something similar to [this example](https://react-datasheet-grid.netlify.app/docs/recipes/tracking-rows-changes) to prevent deletion of a given row

Hope that helps!
